### PR TITLE
[Vue] Refactor router-link to remove tag for v-slot

### LINKED ISSUE #13974: Deprecation warning when using vue
##### **Overview of the issue**

When using vue you can see following deprecation warning in the browser console:

`[vue-router] <router-link>'s tag prop is deprecated and has been removed in Vue Router 4. Use the v-slot API to remove this warning: https://next.router.vuejs.org/guide/migration/#removal-of-event-and-tag-props-in-router-link.`

##### **Motivation for or Use Case**

This looks bad to the user and may be problematicwhen updating vue router or vue.

##### **Reproduce the error**

Follow https://github.com/jhipster/generator-jhipster/issues/13862 and open the application.

##### **Related issues**

None.

##### **Suggest a Fix**

Remove all `tag=` from all `router-link`s. This does not make the warning go away as the whole vue bootstrap navigation (`b-navbar-brand`, `b-nav-item`, `b-nav-item-dropdown`) are using tag and event internally. So we can either leave it like this and wait for a vue bootstrap update or replace their directives/components with hand crafted ones. 

As the [next release with this fix seems not to far away](https://github.com/bootstrap-vue/bootstrap-vue/issues/6373) I would say let's remove the `tag` attribute from our normal router-links and we are good

##### **JHipster Version(s)**

Latest main branch

##### **JHipster configuration**

---
If you have a JDL please wrap it in below structure
  <details>
  <summary>JDL definitions</summary>
  <pre>
application {
  config {
    baseName gateway
    reactive true
    packageName com.okta.developer.gateway
    applicationType gateway
    authenticationType oauth2
    buildTool gradle
    clientFramework vue
    prodDatabaseType postgresql
    serviceDiscoveryType eureka
    testFrameworks [protractor]
  }
  entities Blog, Post, Tag, Product
}

application {
  config {
    baseName blog
    reactive true
    packageName com.okta.developer.blog
    applicationType microservice
    authenticationType oauth2
    buildTool gradle
    databaseType neo4j
    devDatabaseType neo4j
    prodDatabaseType neo4j
    enableHibernateCache false
    serverPort 8081
    serviceDiscoveryType eureka
  }
  entities Blog, Post, Tag
}

application {
  config {
    baseName store
    reactive true
    packageName com.okta.developer.store
    applicationType microservice
    authenticationType oauth2
    buildTool gradle
    databaseType mongodb
    devDatabaseType mongodb
    prodDatabaseType mongodb
    enableHibernateCache false
    serverPort 8082
    serviceDiscoveryType eureka
  }
  entities Product
}

entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}

entity Post {
  title String required
  content TextBlob required
  date Instant required
}

entity Tag {
  name String required minlength(2)
}

entity Product {
  title String required
  price BigDecimal required min(0)
  image ImageBlob
}

relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}

relationship ManyToMany {
  Post{tag(name)} to Tag{post}
}

paginate Post, Tag with infinite-scroll
paginate Product with pagination

microservice Product with store
microservice Blog, Post, Tag with blog

deployment {
  deploymentType docker-compose
  appsFolders [gateway, blog, store]
  dockerRepositoryName "mraible"
}
  </pre>
  </details>

##### **Browsers and Operating System**

Not relevant

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by swarajsaaj**
Refactor router-link tag to v-slot for future support https://next.router.vuejs.org/guide/migration/#removal-of-event-and-tag-props-in-router-link

Fix #13974

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management-view.vue.ejs [hunk 1]
```diff
@@ -53,10 +53,11 @@
             </ul>
           </dd>
         </dl>
-        <router-link tag="button" type="submit"
-                :to="{name: '<%= jhiPrefixCapitalized %>User'}"
-                class="btn btn-info">
+        <router-link custom v-slot="{ navigate }"
+                :to="{name: '<%= jhiPrefixCapitalized %>User'}">
+          <button @click="navigate" type="submit" class="btn btn-info">
```
Comment (atomfrede) [inline generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management-view.vue.ejs:58]: Should it be really a `submit` button and not just a `button` (was already before the case)?

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.vue.ejs [hunk 2]
```diff
@@ -34,7 +36,11 @@
         </thead>
         <tbody v-if ="users">
           <tr v-for="user in users" :key="user.id" :id="user.login">
-            <td><router-link tag="a" :to="{name: '<%= jhiPrefixCapitalized %>UserView', params: {userId: user.login}}">{{user.id}}</router-link></td>
+            <td>
+              <router-link custom v-slot="{ href, navigate }" :to="{name: '<%= jhiPrefixCapitalized %>UserView', params: {userId: user.login}}">
```
Comment (atomfrede) [inline generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.vue.ejs:40]: As far as I understand, an `a` tag is the default, so no need to `custom`ize it here.

**Review by atomfrede — COMMENTED** [review]
Minor comments. Thanks @swarajsaaj for the contribution

**Comment by atomfrede** [comment]
The failing tests are unrelated (ngx)

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management-view.vue.ejs (hunk 1 above)
Comment (swarajsaaj) [reply generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management-view.vue.ejs]: Yes, it was already there, but seems it is not required to be a submit, as its not even a form out there and its just a back functionality. Removed submit in latest commit.

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.vue.ejs (hunk 2 above)
Comment (swarajsaaj) [reply generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.vue.ejs]: Ah yes its not required. Thanks for finding out.

**Pull request merged**